packages/eslint-plugin/src/utils/helpers.ts

```diff
@@ -0,0 +1,11 @@
+import { AST_NODE_TYPES, ASTUtils, TSESTree } from '@typescript-eslint/utils';
+
+type IsHelper<NodeType extends AST_NODE_TYPES> = (node: TSESTree.Node | null | undefined) => node is TSESTree.Node & {
+  type: NodeType;
+};
+
+export const isIdentifier: IsHelper<AST_NODE_TYPES.Identifier> = ASTUtils.isIdentifier;
+export const isObjectExpression: IsHelper<AST_NODE_TYPES.ObjectExpression> = ASTUtils.isNodeOfType(
+  AST_NODE_TYPES.ObjectExpression,
+);
+export const isProperty: IsHelper<AST_NODE_TYPES.Property> = ASTUtils.isNodeOfType(AST_NODE_TYPES.Property);
```

Comment:
Had to type this manually otherwise TypeScript goes wild:

![image](https://user-images.githubusercontent.com/14183168/164479868-ce819feb-c819-4e08-b073-6f3827e72644.png)